docs/reference/ilm/policy-definitions.asciidoc

```diff
@@ -0,0 +1,293 @@
+| `max_age`  | no        | -                   | max duration index exists. See <<time-units, Time Units>> for formatting
+|======
+
+At least one of `max_size`, `max_docs`, `max_age` or any combinations of the three
+are required to be specified.
+
+==== Example: Rollover when index is too large
+
+This example rolls the index over when it is at least 100 gigabytes.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_size": "100GB"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Rollover when index has too many documents
+
+This example rolls the index over when it contains at least
+1000000 documents.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_docs": 1000000
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Rollover when index is too old
+
+This example rolls the index over when it has been created at least
+7 days ago.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_age": "7d"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+==== Example: Rollover when index is too old or too large
+
+This example rolls the index over when it has been created at least
+7 days ago or it is at least 100 gigabytes. In this case, the index will be rolled
+over when at least one of these two conditions is met.
+
+[source,js]
+--------------------------------------------------
+{
+  "rollover" : {
+    "max_age": "7d",
+    "max_size": "100GB"
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+
+[[ilm-shrink-action]]
+=== Shrink
+
+WARNING: This will make the index read-only (see: <<dynamic-index-settings,index.blocks.write>>)
+
+This action shrinks an existing index into a new index with fewer primary shards. It
+calls the <<indices-shrink-index,Shrink API>> to shrink the index. Since allocating
+all the primary shards of the index to one node is a prerequisite, this action will
+first allocate the primary shards to a valid node. After shrinking, it will swap aliases
+pointing to the original index into the new shrunken index. The new index will also
+have a new name: "shrink-<origin-index-name>". So if the original index was called "logs",
+then the new index will be named "shrink-logs".
+
+[[ilm-shrink-options]]
+.Shrink Options
+[options="header"]
+|======
+| Name               | Required  | Default             | Description
+| `number_of_shards` | yes       | -                   | The number of shards to shrink to. must be a factor of the number of shards in the source index.
+|======
+
+[source,js]
+--------------------------------------------------
+{
+  "shrink" : {
+    "number_of_shards": 1
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+
+== Phase Definitions
```

Comment:
I chose to give a high-level overview of which actions are allowed in which phases and deliberately did not mention this information within each action description. I see value in also mentioning this alongside the action description, but didn't know if this would be too unmaintainable moving forward if these change over time.

Reply:
I think this could be combined with the phases section at the top of the document as it feels weird to have the information split with actions in the middle?